spyder_kernels/comms/frontendcomm.py

```diff
@@ -128,25 +67,19 @@ def poll_one(self):
             # use the regular shell stream.
             out_stream = self.kernel.shell_streams[0]
         try:
-            ident, msg = self.kernel.session.recv(self.comm_socket, 0)
+            ident, msg = self.kernel.session.recv(self.kernel.parent.control_socket, 0)
         except zmq.error.ContextTerminated:
             return
         except Exception:
             self.kernel.log.warning("Invalid Message:", exc_info=True)
             return
         msg_type = msg['header']['msg_type']
 
-        if msg_type == 'shutdown_request':
-            self.comm_thread_close.set()
-            self._comm_close(msg)
-            return
-
-        handler = self.kernel.shell_handlers.get(msg_type, None)
+        handler = self.kernel.control_handlers.get(msg_type, None)
         try:
             if handler is None:
                 self.kernel.log.warning("Unknown message type: %r", msg_type)
                 return
-            import asyncio
 
             if (getattr(asyncio, 'run', False) and
                     asyncio.iscoroutinefunction(handler)):
```

Comment:
You can remove this `if` statement and leave only `asyncio.run(handler(out_stream, ident, msg))` because we don't support Python 3.6 anymore.